ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,331 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+@objc(DrawerState)
+
+public class DrawerState: NSObject, ObservableObject {
+    weak var delegate: DrawerStateDelegate?
+    // presentation direction sets the direction for drawer slide out
+    @objc @Published public var presentationDirection: DrawerDirection = .left
+    // state to expand/collpase drawer
+    @objc @Published public var isExpanded: Bool = false {
+        didSet {
+            if isExpanded {
+                delegate?.drawerDidExpand?()
+            } else {
+                delegate?.drawerDidCollapsed?()
+            }
+        }
+    }
+    // if set to true background is dimmed or else by default is clear
+    @objc @Published public var backgroundDimmed: Bool = false
+}
+
+@objc(DrawerStateDelegate)
+
+// State delegate is currently required to notify hosting viewcontroller about expansion
+protocol DrawerStateDelegate: AnyObject {
+    @objc optional func drawerDidExpand()
+    @objc optional func drawerDidCollapsed()
+}
+
+public class DrawerTokens: ObservableObject {
+
+    @Published public var shadowColor: Color!
+    @Published public var shadowOpacity: Double!
+    @Published public var shadowBlur: CGFloat!
+    @Published public var shadowDepth: [CGFloat]! // x,y axis
+
+    public init() {
+        self.themeAware = true
+        didChangeAppearanceProxy()
+    }
+
+    @objc open func didChangeAppearanceProxy() {
+        let appearanceProxy = StylesheetManager.S.DrawerTokens
+        shadowColor = Color(appearanceProxy.shadowColor)
+        shadowOpacity = Double(appearanceProxy.shadowOpacity)
+        shadowBlur = appearanceProxy.shadowBlur
+        shadowDepth = appearanceProxy.shadowDepth
+    }
+}
+
+// Mark - Drawer View
+
+@objc public enum DrawerDirection: Int, CaseIterable {
+    case left
+    case right
+}
+
+/**
+ `Drawer` is used to present a content that is currently in hideout mode when collapsed and when expanded is used to present content view partially either on the left or right.
+ 
+ Currently `Drawer` only support horizontal axis and is expanded by default from left side of the screen unless explicitly specified
+ 
+ Set `Content` to provide content for the drawer.
+ */
+public struct Drawer<Content: View>: View {
+
+    @ObservedObject var state: DrawerState
+    @ObservedObject var tokens: DrawerTokens
+
+    private let backgroundLayerColor: Color = .black
+    private var backgroundLayerOpacity: Double {
+        return state.backgroundDimmed ? 0.5 : 0
+    }
+    private let backgroundLayerAnimation: Animation = .default
+
+    private let percentWidthOfContent: CGFloat = 0.9
+    private let percentSnapWidthOfScreen: CGFloat = 0.225
+
+    // content view on top of
```

Comment:
Is this comment incomplete?

Reply:
Thanks for catching this, I have updated it